sdk/servicebus/service-bus/src/serviceBusClient.ts

```diff
@@ -242,15 +242,17 @@ export class ServiceBusClient {
    * Creates a Sender which can be used to send messages, schedule messages to be sent at a later time
    * and cancel such scheduled messages.
    */
-  createSender(queueOrTopicName: string): Sender {
+  async createSender(queueOrTopicName: string): Promise<Sender> {
     validateEntityNamesMatch(this._connectionContext.config.entityPath, queueOrTopicName, "sender");
 
     const clientEntityContext = ClientEntityContext.create(
       queueOrTopicName,
       this._connectionContext,
       `${queueOrTopicName}/${generate_uuid()}`
     );
-    return new SenderImpl(clientEntityContext, this._clientOptions.retryOptions);
+    const sender = new SenderImpl(clientEntityContext, this._clientOptions.retryOptions);
+    await sender.open();
```

Comment:
So with this change, if the sender can't be opened, then the user doesn't get a sender at all and instead get an error right? Is that intentional?

Reply:
The original motivation for this was to front-load the work but we had another issue where people wanted a deterministic way to make sure that the connection was actually loaded up and running.

So yes, intentional and you're right, they won't be able to get a sender.

Reply:
>  people wanted a deterministic way to make sure that the connection was actually loaded up and running
Is this meant only for when the sender is 1st created? I ask since it's possible for the connection to be lost at any point after the sender is created.

Reply:
This is only meant for when the sender is first created.

However, you brought up a good point that lazy connect would often use retries. So that has now been added :)